Pod/Classes/PINProgressiveImage.m

```diff
@@ -350,59 +301,135 @@ - (CFTimeInterval)estimatedRemainingTime
 }
 
 //Must be called within lock
+//Heavily cribbed from https://developer.apple.com/library/ios/samplecode/UIImageEffects/Listings/UIImageEffects_UIImageEffects_m.html#//apple_ref/doc/uid/DTS40013396-UIImageEffects_UIImageEffects_m-DontLinkElementID_9
 - (UIImage *)postProcessImage:(UIImage *)inputImage withProgress:(float)progress
 {
+    UIImage *outputUIImage = nil;
     CGImageRef inputImageRef = CGImageRetain(inputImage.CGImage);
     if (inputImageRef == nil) {
         return nil;
     }
     
-    CIFilter *gaussianFilter = [CIFilter filterWithName:@"CIGaussianBlur"];
-    [gaussianFilter setDefaults];
-    
-    UIImage *outputUIImage = nil;
-    
-    CIContext *processingContext = GPUProcessingContext;
-    if (self.inBackground) {
-        processingContext = CPUProcessingContext;
-    }
-    
-    CGSize maxInput = processingContext.inputImageMaximumSize;
-    CGSize maxOutput = processingContext.outputImageMaximumSize;
+    CGSize maxSize = CGSizeMake(1024, 1024);
     CGSize inputSize = inputImage.size;
-    if (maxInput.height < inputSize.height ||
-        maxInput.width < inputSize.width ||
-        maxOutput.height < inputSize.height ||
-        maxOutput.width < inputSize.width) {
+    if (inputSize.width < 1 ||
+        inputSize.height < 1 ||
+        maxSize.height < inputSize.height ||
+        maxSize.width < inputSize.width) {
         CGImageRelease(inputImageRef);
         return nil;
     }
     
-    if (processingContext && gaussianFilter) {
-        [gaussianFilter setValue:[CIImage imageWithCGImage:inputImageRef]
-                          forKey:kCIInputImageKey];
+    CGFloat radius = MAX(__FLT_EPSILON__, (inputImage.size.width / 25.0) * MAX(0, 1.0 - progress));
```

Comment:
Perhaps it's reasonable for this value to reach 0, and then just skip the blur if the calculation ever yielded that, instead of __FLT_EPSILON__?